miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -478,6 +517,21 @@ private void setIdentity(DetailedSample descendant) throws IOException, MisoNami
     child.setParent(identity);
   }
 
+  /**
+   * Finds the identity of a sample by climbing the hierarchy
+   * 
+   * @param sample
+   * @return identity attached to sample or null if no identity is attached
+   * @throws IOException
+   */
+  private SampleIdentity getIdentity(DetailedSample sample) throws IOException {
+    if (sample.getParent() == null) {
+      throw new IllegalArgumentException(ERR_MISSING_PARENT_ID);
+    }
+    SampleIdentity identity = LimsUtils.getParent(SampleIdentity.class, sample);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "identity". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)